### PR TITLE
policy: Treat taproot as always active

### PR DESCRIPTION
**Pull request by maflcko**
Now that taproot is active, it can be treated as if it was always active for policy for the next major release. This simplifies the code and changes two things:

* Importing `tr` descriptors can be done before the chain is fully synced. This is fine, because the wallet will already generate `tr` descriptors by default (regardless of the taproot status) after commit 47fe7445e7f54aee10ec6dbc53f1db1adbeb43de.
* Valid taproot spends won't be rejected from the mempool before taproot is active. This is strictly speaking a bugfix after commit 47fe7445e7f54aee10ec6dbc53f1db1adbeb43de, since the wallet may generate taproot spends before the chain is fully synced. For example, a slow node or a purposefully offline node. Currently, the wallet needs the mempool to account for change. See https://github.com/bitcoin/bitcoin/issues/11887.

A similar change was done for segwit v0 in https://github.com/bitcoin/bitcoin/pull/13120 .

This effectively reverts commit c5ec0367d718544caa3a1578d6c730fc92ee4e94.

### CODE REVIEW

**Comment by sipa** [comment]
Concept ACK

File: test/functional/feature_taproot.py [hunk 1]
```diff
@@ -435,8 +436,11 @@ def to_script(elem):
 # - Whether this test demands being placed in a txin with no corresponding txout (for testing SIGHASH_SINGLE behavior)
 
 Spender = namedtuple("Spender", "script,comment,is_standard,sat_function,err_msg,sigops_weight,no_fail,need_vin_vout_mismatch")
+# The full node versions treat it standard V23 means the major version 23.0
```
Comment (achow101) [inline test/functional/feature_taproot.py:439]: This comment is hard to parse, and it doesn't really explain each value in the enum. Could you please clarify?

**Comment by MarcoFalke** [comment]
@achow101 Sorry, I accidentally deleted your comment. It was a comment on the docstring of `Standard`:

> This comment is hard to parse, and it doesn't really explain each value in the enum. Could you please clarify?

Thanks, I've force pushed a new docstring.

**Comment by mjdietzx** [comment]
Code Review ACK fa3e0da06b491b8c0fa2dbae37682a9112c9deb8

**Comment by achow101** [comment]
ACK fa3e0da06b491b8c0fa2dbae37682a9112c9deb8

**Comment by sipa** [comment]
utACK fa3e0da06b491b8c0fa2dbae37682a9112c9deb8

**Comment by gruve-p** [comment]
ACK https://github.com/bitcoin/bitcoin/pull/23512/commits/fa3e0da06b491b8c0fa2dbae37682a9112c9deb8

File: test/functional/feature_taproot.py [hunk 2]
```diff
@@ -435,8 +436,14 @@ def to_script(elem):
...
 
 Spender = namedtuple("Spender", "script,comment,is_standard,sat_function,err_msg,sigops_weight,no_fail,need_vin_vout_mismatch")
+# The full node versions that treat the tx standard.
+# ALL means any version
+# V23 means the major version 23.0 and any later version
+# NONE means no version
+Standard = Enum('Standard', 'ALL V23 NONE')
 
-def make_spender(comment, *, tap=None, witv0=False, script=None, pkh=None, p2sh=False, spk_mutate_pre_p2sh=None, failure=None, standard=True, err_msg=None, sigops_weight=0, need_vin_vout_mismatch=False, **kwargs):
+
```
Comment (rajarshimaitra) [inline test/functional/feature_taproot.py:445]: This seems to be an extra line of space.

**Review by rajarshimaitra — COMMENTED** [review]
code review + tACK https://github.com/bitcoin/bitcoin/pull/23512/commits/fa3e0da06b491b8c0fa2dbae37682a9112c9deb8

File: test/functional/feature_taproot.py (hunk 2 above)
Comment (MarcoFalke) [reply test/functional/feature_taproot.py:445]: According to pep-8, two lines are recommended.

**Comment by gunar** [comment]
Code Review + tACK fa3e0da06
